binutils.sh

```diff
@@ -4,55 +4,55 @@ BINUTILS_DL="https://ftp.gnu.org/gnu/binutils"
 BINUTILS_XZ="binutils-2.38.tar.xz"
 BINUTILS_DIR="$(basename $BINUTILS_XZ .tar.xz)"
 BINUTILS_URL="$BINUTILS_DL/$BINUTILS_XZ"
-ORIG_DIR="$(pwd)"
+ORIGIN_DIR="$(CDPATH= cd -- "$(dirname -- "$0")" && pwd)"
 
 # Dont continue on error.
 set -e
 
 # Help text.
 if [ "$1" = '-h' ]; then
-   echo >&2 "Usage: $0 [clean]"
-   echo >&2
-   echo >&2 'Specify "clean" to remove binutils, otherwise it will be downloaded and built.'
-   exit 1
+    echo >&2 "Usage: $0 [clean]"
+    echo >&2
+    echo >&2 'Specify "clean" to remove binutils, otherwise it will be downloaded and built.'
+    exit 1
 fi
 
 # Optional cleanup if requested.
 if [ "$1" = 'clean' ]; then
-   rm -fv objcopy objdump ld "$BINUTILS_XZ"
-   rm -rfv "$BINUTILS_DIR"
-   exit
+    rm -fv "$ORIGIN_DIR/objcopy" "$ORIGIN_DIR/objdump" "$ORIGIN_DIR/ld"
+    rm -rfv "$BINUTILS_DIR"
+    rm -fv "$ORIGIN_DIR/$BINUTILS_XZ"
+    exit
 fi
 
 # Download binutils.
 if [ ! -f "$BINUTILS_XZ" ]; then
-   wget "$BINUTILS_URL"
+    wget "$BINUTILS_URL"
 fi
 
 # Extract binutils.
 if [ ! -d "$BINUTILS_DIR" ]; then
-   tar xf "$BINUTILS_XZ"
+    xzcat "$BINUTILS_XZ" | tar xf -
 fi
 
 # Compile binutils.
 if [ ! -x "$BINUTILS_DIR/binutils/objcopy" ]; then
-   cd "$BINUTILS_DIR"
-   ./configure --enable-targets=i386-pc-elf32 \
-               --disable-gas                  \
-               --disable-libctf               \
-               --disable-plugins              \
-               --disable-gprof                \
-               --enable-compressed-debug-sections=none
-
-   make all-ld -j$(nproc) MAKEINFO=true
+    cd "$BINUTILS_DIR" && \
+    ./configure --enable-targets=i386-pc-elf32 \
+                --disable-gas \
+                --disable-libctf \
+                --disable-plugins \
+                --disable-gprof \
+                --enable-compressed-debug-sections=none && \
+    gmake all-ld -j$(nproc) MAKEINFO=true
```

Comment:
Command `gmake` is not found in default Ubuntu Linux installation even after running `sudo apt install make`. Can you specify the appropriate command (`gmake` or `make`) by detecting at runtime?

Reply:
Most certainly. Added a script to detect GNU make.